taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -0,0 +1,1034 @@
+			fmt.Println("Unable to parse cancelUndelegate input args: ", err)
+			return nil, err
+		}
+
+		return nil, self.cancelUndelegate(ctx, evm.GetBlock().Number, args)
+
+	case "reDelegate":
+		var args RedelegateArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse reDelegate input args: ", err)
+			return nil, err
+		}
+
+		return nil, self.redelegate(ctx, evm.GetBlock().Number, args)
+
+	case "claimRewards":
+		var args ValidatorAddress
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse claimRewards input args: ", err)
+			return nil, err
+		}
+
+		return nil, self.claimRewards(ctx, evm.GetBlock().Number, args)
+
+	case "claimCommissionRewards":
+		return nil, self.claimCommissionRewards(ctx, evm.GetBlock().Number)
+
+	case "registerValidator":
+		var args RegisterValidatorArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse registerValidator input args: ", err)
+			return nil, err
+		}
+
+		return nil, self.registerValidator(ctx, evm.GetBlock().Number, args)
+
+	case "setValidatorInfo":
+		var args SetValidatorInfoArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse setValidatorInfo input args: ", err)
+			return nil, err
+		}
+
+		return nil, self.setValidatorInfo(ctx, args)
+
+	case "isValidatorEligible":
+		var args ValidatorAddress
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse isValidatorEligible input args: ", err)
+			return nil, err
+		}
+		result := self.delayedStorage.IsEligible(&args.Validator)
+		return method.Outputs.Pack(result)
+
+	case "getTotalEligibleValidatorsCount":
+		result := self.delayedStorage.EligibleAddressCount()
+		return method.Outputs.Pack(result)
+
+	case "getTotalEligibleVotesCount":
+		result := self.delayedStorage.EligibleVoteCount()
+		return method.Outputs.Pack(result)
+
+	case "getValidatorEligibleVotesCount":
+		var args ValidatorAddress
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse getValidatorEligibleVotesCount input args: ", err)
+			return nil, err
+		}
+
+		result := self.delayedStorage.GetEligibleVoteCount(&args.Validator)
+		return method.Outputs.Pack(result)
+
+	case "getValidators":
+		var args GetValidatorsArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse getValidators input args: ", err)
+			return nil, err
+		}
+
+		result := self.getValidators(args)
+		return method.Outputs.Pack(result)
+
+	case "getDelegatorDelegations":
+		var args GetDelegatorDelegationsArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse getDelegatorDelegations input args: ", err)
+			return nil, err
+		}
+
+		result := self.getDelegatorDelegations(args)
+		return method.Outputs.Pack(result)
+
+	case "getUndelegations":
```

Comment:
This should be somehow merged with getDelegatorDelegations method. Staking website will fetch data for specific delegator and it should return all delegations & undelegation request for all validators that delegator interacted with.

Reply:
I disagree, normal website does not require undelegation data, so we should make that call even more expensive

Reply:
... nice that you disagree. When you login, you wanna see all of your delegations so you can claim rewards by clicking on button for concrete validator + you wanna see also all undelegations so you can click "confirm". So tell me how you wanna solve this if the website does not require this data ?

Reply:
maybe @rattrap should join our discussion as he has experience with FE

Reply:
If Leo is ok with calling 2 separate functions then I guess whatever. If they want to call only 1 function then we should be able to serve merged data